systemd/systemd.go

```diff
@@ -1091,9 +1143,25 @@ LazyUnmount=yes
 WantedBy=multi-user.target
 `
 
-func writeMountUnitFile(snapName, revision, what, where, fstype string, options []string) (mountUnitName string, err error) {
-	content := fmt.Sprintf(mountUnitTemplate, snapName, revision, what, where, fstype, strings.Join(options, ","))
-	mu := MountUnitPath(where)
+const (
+	snappyOriginModule = "X-SnapdOrigin"
+)
+
+func writeMountUnitFile(u *MountUnitOptions) (mountUnitName string, err error) {
+	if u == nil {
+		panic("writeMountUnitFile() expects valid mount options")
+	}
+
+	snapDesc := u.SnapName
+	if u.Revision != "" {
+		snapDesc += fmt.Sprintf(", revision %s", u.Revision)
+	}
```

Comment:
as I mentioned in the big PR I think the unit description should incorporate Origin if it's set:

Mount unit for SNAP via ORIGIN

or something like that

Reply:
Done!